vagrant/tournament/tournament_test.py

```diff
@@ -99,41 +92,64 @@ def testReportMatches():
+        if w != 0:
+            raise ValueError("After deleting matches, players should have zero wins recorded.")
+    print "8. After match deletion, player standings are properly reset.\n9. Matches are properly deleted."
 
 def testPairings():
+    """
+    Test that pairings are generated properly both before and after match reporting.
+    """
     deleteMatches()
     deletePlayers()
     registerPlayer("Twilight Sparkle")
     registerPlayer("Fluttershy")
     registerPlayer("Applejack")
     registerPlayer("Pinkie Pie")
+    registerPlayer("Rarity")
+    registerPlayer("Rainbow Dash")
+    registerPlayer("Princess Celestia")
+    registerPlayer("Princess Luna")
     standings = playerStandings()
-    [id1, id2, id3, id4] = [row[0] for row in standings]
+    [id1, id2, id3, id4, id5, id6, id7, id8] = [row[0] for row in standings]
+    pairings = swissPairings()
+    if len(pairings) != 4:
+        raise ValueError(
+            "For eight players, swissPairings should return 4 pairs. Got {pairs}".format(pairs=len(pairings)))
     reportMatch(id1, id2)
     reportMatch(id3, id4)
+    reportMatch(id5, id6)
+    reportMatch(id7, id8)
     pairings = swissPairings()
-    if len(pairings) != 2:
+    if len(pairings) != 4:
         raise ValueError(
-            "For four players, swissPairings should return two pairs.")
-    [(pid1, pname1, pid2, pname2), (pid3, pname3, pid4, pname4)] = pairings
-    correct_pairs = set([frozenset([id1, id3]), frozenset([id2, id4])])
-    actual_pairs = set([frozenset([pid1, pid2]), frozenset([pid3, pid4])])
-    if correct_pairs != actual_pairs:
-        raise ValueError(
-            "After one match, players with one win should be paired.")
-    print "8. After one match, players with one win are paired."
+            "For eight players, swissPairings should return 4 pairs. Got {pairs}".format(pairs=len(pairings)))
+    [(pid1, pname1, pid2, pname2), (pid3, pname3, pid4, pname4), (pid5, pname5, pid6, pname6), (pid7, pname7, pid8, pname8)] = pairings
+    possible_pairs = set([frozenset([id1, id3]), frozenset([id1, id5]),
+                          frozenset([id1, id7]), frozenset([id3, id5]),
+                          frozenset([id3, id7]), frozenset([id5, id7]),
+                          frozenset([id2, id4]), frozenset([id2, id6]),
+                          frozenset([id2, id8]), frozenset([id4, id6]),
+                          frozenset([id4, id8]), frozenset([id6, id8])
+                          ])
+    actual_pairs = set([frozenset([pid1, pid2]), frozenset([pid3, pid4]), frozenset([pid5, pid6]), frozenset([pid7, pid8])])
+    for pair in actual_pairs:
+        if pair not in possible_pairs:
+            raise ValueError(
+                "After one match, players with one win should be paired.")
+    print "10. After one match, players with one win are properly paired."
 
 
 if __name__ == '__main__':
-    testDeleteMatches()
```

Comment:
It's a minor nit, and Karl agrees that they're somewhat redundant. I'm okay merging this now since these new tests are already an obvious improvement. If you have a nifty way to include these tests too we can merge those in later :+1: